src/core/message.ts

```diff
@@ -105,32 +201,85 @@ export interface Message {
  * Allows for comparison of messages based on their timeserials.
  */
 export class DefaultMessage implements Message {
-  private readonly _calculatedTimeserial: Timeserial;
+  private readonly _calculatedOriginTimeserial: Timeserial;
+  private readonly _calculatedActionSerial: Timeserial;
+  public readonly createdAt: Date;
 
   constructor(
     public readonly timeserial: string,
     public readonly clientId: string,
     public readonly roomId: string,
     public readonly text: string,
-    public readonly createdAt: Date,
     public readonly metadata: MessageMetadata,
     public readonly headers: MessageHeaders,
+
+    public readonly latestAction: ChatMessageActions,
+    // the latestActionSerial will be set to the original timeserial for new messages,
+    // else it will be set to the serial corresponding to whatever action
+    // (update/delete) that was just performed.
+    public readonly latestActionSerial: string,
+
+    public readonly deletedAt?: Date,
+    public readonly updatedAt?: Date,
+    public readonly latestActionDetails?: MessageActionDetails,
   ) {
-    this._calculatedTimeserial = DefaultTimeserial.calculateTimeserial(timeserial);
+    this._calculatedOriginTimeserial = DefaultTimeserial.calculateTimeserial(timeserial);
+    this._calculatedActionSerial = DefaultTimeserial.calculateTimeserial(latestActionSerial);
+    this.createdAt = new Date(this._calculatedOriginTimeserial.timestamp);
```

Comment:
Considering we will be moving to lexico serials, I think we should avoid this and instead rely on the original timestamp, wdyt?

Reply:
Don't lexico serials still have a timestamp in?

Reply:
Are we OK to follow up with this change?

So leave this PR as-is to show correct values, and when we change to lexico (soon) we remove the parsing and re-add the createdAt from timestamp. By then the timestamp fix will also be fixed+merged?

Reply:
Yeh that's fine with me